src/server/pfs/server/testing/server_test.go

```diff
@@ -7614,3 +7613,28 @@ func TestNilBranchNameStream(t *testing.T) {
 		return nil
 	}))
 }
+
+func TestWalkFileInvalid(t *testing.T) {
+	ctx := pctx.TestContext(t)
+	env := realenv.NewRealEnv(ctx, t, dockertestenv.NewTestDBConfig(t).PachConfigOption)
+
+	repo := "test"
+	require.NoError(t, env.PachClient.CreateRepo(pfs.DefaultProjectName, repo))
+
+	commit1, err := env.PachClient.StartCommit(pfs.DefaultProjectName, repo, "master")
+	require.NoError(t, err)
+	require.NoError(t, env.PachClient.PutFile(commit1, "/a", &bytes.Buffer{}))
+	require.NoError(t, finishCommit(env.PachClient, repo, "", commit1.Id))
+
+	commit2, err := env.PachClient.StartCommit(pfs.DefaultProjectName, repo, "master")
+	require.NoError(t, err)
+	require.NoError(t, env.PachClient.PutFile(commit2, "/a/a", &bytes.Buffer{}))
+
+	_, err = env.PachClient.InspectFile(commit2, "/")
+	require.YesError(t, err)
+
+	require.NotNil(t, env.PachClient.WalkFile(commit2, "/", func(fi *pfs.FileInfo) error {
```

Comment:
This part of the test seems like it can be removed.

Reply:
Do you mean remove WalkFile?

Reply:
Yeah, but it might just be better restructure it. I would change this to a `YesError` check and the callback function to just be a no-op (i.e. just return nil).